pkg/sysinfo: omit Getpid call in numCPU

tklauser opened the pull request:
The man page for sched_setaffinity(2) states the following about the pid argument [1]:

> If pid is zero, then the mask of the calling thread is returned.

Thus the additional call to unix.Getpid can be omitted and pid = 0 passed to unix.SchedGetaffinity.

[1] https://man7.org/linux/man-pages/man2/sched_setaffinity.2.html#DESCRIPTION

thaJeztah reviewed (approved):
LGTM

AkihiroSuda commented on pkg/sysinfo/numcpu_linux.go:
```diff
@@ -1,23 +1,18 @@
 // for use to this process.
 //
-// Issues two syscalls.
+// Issues one syscall.
```
This line can be just removed

tklauser committed:
pkg/sysinfo: omit Getpid call in numCPU

The man page for sched_setaffinity(2) states the following about the pid
argument [1]:

> If pid is zero, then the mask of the calling thread is returned.

Thus the additional call to unix.Getpid can be omitted and pid = 0
passed to unix.SchedGetaffinity.

[1] https://man7.org/linux/man-pages/man2/sched_setaffinity.2.html#DESCRIPTION

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
